### ISSUE TITLE
external image is darker than it should be due to 16384 => 65535 scaling

### ISSUE THREAD

**Issue by chiuhc** [yuanming-hu/fc4#24]
Hi, 
It's a great reading experience and a lot of fun to play with the code. One question.

In test() function under fc4.py you apply 16384 for 14 bit DSLR,     
　img = (img / 255.0) ** 2.2 * 16384
but when you visualize the image you apply 65535 in get_visualization() under summary_utils.py    
　img = tf.pow(images[:, :, :, ::-1] / 65535, 1 / VIS_GAMMA)
　img_corrected = tf.pow(images[:, :, :, ::-1] / 65535 / illums_pooled[:, None, None, :] *
          tf.reduce_mean(illums_pooled, axis=(1), keep_dims=True)[:, None, None, :],      1 / VIS_GAMMA)

This makes the "input panel", i.e. upper-left of your 6 panel figure obviously darker than it should be,
particularly when you test external figures and compare the input png with the 6 panel output.
Can you explain why you do this different scaling?

**Comment by yuanming-hu** [comment]
Hi @chiuhc , I think you are right and I should have used 65536 here. I fixed it in 9c16174.
Thank you for point this out.

**Comment by chiuhc** [comment]
Hi,  
Thanks for prompt response. I wanna double confirm: does it mean you used 65535 as dynamic range when training the the fc4 model. (As you apply 65535 in the dataset.py). And I should play with the 9c16174 commit again, ignoring the 16384 results I generated for external image, because 16384 scaling may produce worse illums_pooled for external image? 

dataset.py
　def load_image_without_mcc(self, r):
　　raw = self.load_image(r.fn)
　　img = (np.clip(raw / raw.max(), 0, 1) * 65535.0).astype(np.uint16)

**Comment by yuanming-hu** [comment]
Yes, please use 9c16174 for external images. Using 16384 may make your outputs look darker. When I was developing this project I never used jpg or png as inputs since my data were all 16-bit raw. Sorry for the inconvenience.
